### DISCUSSION TITLE
Uefi Patch required?

### DISCUSSION THREAD

**Discussion by DJGroovin** [terminatorul/NvStrapsReBar#102] [General]
Hello all, I seem a little confused and i may not have understood the instructions or I have read too much into it and have confused myself...

I have a Z590 motherboard, ReBAR is supported natively on this board.  I have just gotten my hands on a EVGA 2080 ti.  Do I need to patch the bios still, or is this step not required since the board supports it already?

I was under the impression that I just need to enable it in the board, and run the nvstrap executable, but didnt want to screw it up.  

any help is apprecaited, and i apologize if its been asked, but I was getting conflicting info oon search results.

Thanks again

**Comment by D337z** [comment]
Yes, in order to use ReBAR on the 2080 Ti, you would need to patch the UEFI.  This is because the UEFI doesn't attempt to use ReBAR with the card by default.  Only with the patch can you force a specific BAR size for the PCI-E to use with the card.  It's not so much resizable, though, as much as settable.  Should you upgrade your GPU in the future, make sure to disable NVStraps before installing it.

**Reply by DJGroovin** [reply]
Thanks for the reply.

So UEFU patch required and NVStraps required....check and also need the nvidia game profiler too right?

What has the experience been with actual performance gains? Has there been any crashes or odd behaviors?

**Comment by D337z** [comment]
To make it work, you must patch the UEFI of your motherboard with the NVStraps file.  Then, make sure that you have your system in UEFI mode, enable 4G decoding, and disable CSM.
Then you'll use the NVStraps exe program to set a specific size that is larger than the default and then reboot.
You can see some FPS improvement after changing the size, but results may vary. 
Please note: Modifying the UEFI could invalidate future UEFI updates on some boards making it difficult to update on the future.

**Reply by DJGroovin** [reply]
What's this about nvidia profiler...is that required as well?

**Comment by D337z** [comment]
The profiler is to set what applications actually should take advantage of
the larger BAR size.  Not all applications work well with larger BAR.

**Reply by DJGroovin** [reply]
So you have to.enable it manually for every game inside the profiler?

**Comment by D337z** [comment]
More-so disable it for each old game that doesn't work well with it, but
same idea.

**Reply by DJGroovin** [reply]
Is this profiler app necessary for a 3000 or 4000 series card that has rebar support available by default? It's the first timeni have seen the profilernapp so just curious

**Comment by D337z** [comment]
If you're using a 3000 or 4000 series, you'll want to use ReBAR instead.
The profiler will still be useful.  You'll also need to use the vBIOS
update for the 3000 series cards usually.

**Reply by DJGroovin** [reply]
Ok...I hit a snag...

I'm have managed to open the bios with uefitool and add the ffs patch.  When I open uefipatch and run the command, I get:

No patches can be applied to input file

It then says I should use dsdt patching which when you click on its instructions, I'm not one if the cases...  should I use mmtool instead of uefitool? Am I doing something wrong?

My motherboard is an asus tuf z590 gaming plus wifi.

Thanks for your help on this

**Comment by D337z** [comment]
Hold on, if you're using an RTX3000+ GPU, Resizable BAR is already built in
to your motherboard and should be fairly easy to enable without the need
for am additional module.

**Reply by DJGroovin** [reply]
I have a EVGA 2080 Ti, I just got it as a warranty replacement for my old evga 1080.  I was in the market looking to buy a 4070 prior to evga sending me the 2080 ti...and it has all the power I need for now...  my sons machine runs a 1650 on an asus x99 motherboard so I can technically use thus on both systems....just seem to be stuck at this point...

**Comment by D337z** [comment]
I'm that case, this would work.  But you'll have to use the other tool as
it says.  Updating the UEFI to your motherboard will be tricky though.

**Comment by DJGroovin** [comment]
I can't seem to get around the error...

I'm not sure if I'm doing something wrong...I tried using mmtool but the pcibus isn't labelled there and I can't locate it to even try that method...

**Comment by DJGroovin** [comment]
I just tried doing it on my asus b550-f wifi bios and during the uefitool part, there were 2 entries found so I entered the nvstraps module twice as required...but once again during uefipatch I'm getting the same error...

Should I be using the patch .txt file from.the github about adding the ffs module or the default on that's in the directory of uefipatch when you unzip it...I know these are both asus boards, but others seem to have succeeded doing the patch and I'm totally stuck....do you have any idea why this could be happening??

**Comment by D337z** [comment]
I'm not sure that you'll even be required to patch the UEFI with anything
other than the ffs module.  However, what may bee happening would be that
the UEFI is beyond the version supported by the MMTool hence the issue.  I
could be wrong and I don't fully understand what error you're seeing, but
it really should work fine.

**Reply by DJGroovin** [reply]
![Screenshot 2024-09-03 235650](https://github.com/user-attachments/assets/070e0668-9dae-4464-9be6-9697c6810b8d)

thats the error i keep getting no matter what i try....
my motherboard is the asus TUF Z590 Gaming plus Wifi, using bios 1801. incase you wanted to take a closer peak at what could be causing this...  

I have an older asus x99 deluxe laying around, I may attempt this with the bios for that board just as a test to see if i can get the biospatch portion of this completed....unfortuantely that board has had resume from sleep problems for the longest time, so its just used as a test bench board for me now.....Its not overclocked other than ram xmp settings, and even there, the ram is on the vendors QVL list....anyhow...that X99 system isnt the issue right now....

**Comment by D337z** [comment]
Ah. It's a result of not following the instructions carefully enough.  It's
UEFIpatch {name of txt file with patches} {Name of ROM file}
Without the {}
You'll likely need to redownload the UEFI file since you seem to have
patched it with itself or something.

**Reply by DJGroovin** [reply]
I will try that.... someone sent me a youtube video showing it done line this so i followed it.....
![image](https://github.com/user-attachments/assets/66c22464-bc4b-4a71-b6fc-356248541e85)

**Reply by DJGroovin** [reply]
> Ah. It's a result of not following the instructions carefully enough. It's UEFIpatch {name of txt file with patches} {Name of ROM file} Without the {} You'll likely need to redownload the UEFI file since you seem to have patched it with itself or something.
> […](#)
> On Wed, Sep 4, 2024, 11:36 DJGroovin ***@***.***> wrote: Screenshot.2024-09-03.235650.png (view on web) <https://github.com/user-attachments/assets/070e0668-9dae-4464-9be6-9697c6810b8d> thats the error i keep getting no matter what i try.... my motherboard is the asus TUF Z590 Gaming plus Wifi, using bios 1801. incase you wanted to take a closer peak at what could be causing this... I have an older asus x99 deluxe laying around, I may attempt this with the bios for that board just as a test to see if i can get the biospatch portion of this completed....unfortuantely that board has had resume from sleep problems for the longest time, so its just used as a test bench board for me now.....Its not overclocked other than ram xmp settings, and even there, the ram is on the vendors QVL list....anyhow...that X99 system isnt the issue right now.... — Reply to this email directly, view it on GitHub <[#102 (reply in thread)](https://github.com/terminatorul/NvStrapsReBar/discussions/102#discussioncomment-10548475)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ATOTXIZWKHD5QAZZWYRYW5DZU4ZHRAVCNFSM6AAAAABMWR4HXWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJUHA2DONI> . You are receiving this because you commented.Message ID: ***@***.*** com>

unfortunately, same result.... typing in just uefipatch will show parameteres and it does say that it uses patches.txt by default 

this does not seem to be the root cause....

im either doing something wrong, or theres a bug....im not ruling out that im making a mistake, but i cant seem to track where that could happen, i mean the only other thing is where the dxe driver is being inserted but im following those instructions too.....

**Comment by D337z** [comment]
I would suggest going back and following the instructions.

**Reply by DJGroovin** [reply]
![Screenshot 2024-09-04 141924](https://github.com/user-attachments/assets/7642bddd-7651-4218-975c-84d196d623f8)

Hi D337z, Thanks for your help.

i am following the instructions.... if you seen the screen shot from the rebar patching ffs page, its identical....  it definitely was something to try, but its still the same result....

I just tried modifying the bios for my testbench board, a x99 deluxe and uefipatch patches the bios without spitting out the error....  theres something else going on.

**Comment by DJGroovin** [comment]
@terminatorul 

Hi,

Im curious if you have any insights on this....I'd like to enable this, however am at a crossroads....I believe i have followed the instructions but am getting this error as you can see in the screenshots above.  Any thoughts or advice would greatly be appreciated?

**Comment by D337z** [comment]
Try just adding the module instead of trying to patch it.
